models/user.go

```diff
@@ -0,0 +1,35 @@
+package models
+
+import (
+	"github.com/beego/beego/v2/client/orm"
+)
+
+type User struct {
+	Default
+
+	Email             string `orm:"unique;size(128)"`
+	EncryptedPassword string
+}
+
+func init() {
+	orm.RegisterModel(new(User))
+}
+
+// AddUser insert a new User into database and returns
+// last inserted Id on success.
+func AddUser(m *User) (id int64, err error) {
+	o := orm.NewOrm()
+	id, err = o.Insert(m)
+	return
+}
+
+// GetUserByEmail retrieves User by Email. Returns error if
+// Email doesn't exist
+func GetUserByEmail(email string) (v *User, err error) {
+	o := orm.NewOrm()
+	v = &User{Email: email}
+	if err = o.QueryTable(new(User)).Filter("Email", email).RelatedSel().One(v); err == nil {
```

Comment:
```suggestion
err := o.QueryTable(new(User)).Filter("Email", email).RelatedSel().One(v)
if err == nil {...}
```
According to our [convention](https://nimblehq.co/compass/development/code-conventions/golang/#errors)

and IMO, it confuse when to check if error is nil as most of the Go code check for error is not nil. How about this

```go
if err != nil {
   return nil, err
}
return v, nil
```